src/components/implementation/no_interface/scheddev/sl_mod_fprr.c

```diff
@@ -70,10 +72,32 @@ sl_mod_thd_delete(struct sl_thd_policy *t)
 void
 sl_mod_thd_param_set(struct sl_thd_policy *t, sched_param_type_t type, unsigned int v)
 {
-	assert(type == SCHEDP_PRIO && v < SL_FPRR_NPRIOS);
-	ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
-	t->priority = v;
-	ps_list_head_append_d(&threads[t->priority], t);
+	switch (type) {
+	case SCHEDP_PRIO:
+	{
+		assert(v < SL_FPRR_NPRIOS);
+		ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
+		t->priority = v;
+		ps_list_head_append_d(&threads[t->priority], t);
+		sl_thd_setprio(sl_mod_thd_get(t), t->priority);
+
+		break;
+	}
+	case SCHEDP_WINDOW:
```

Comment:
This is broken as it is bunding the scheduling policy with the timeout policy.

However, I'm not convinced we'll *ever* use a different timeout module.  I'm not sure why I thought this in the initial design.  It seems like the timeout stuff should just be a core part of the scheduling library.  In that case, `SCHEDP_WINDOW` should likely be interpreted by the wakeup/timeout policy, perhaps in addition to the code here.